demo/transient/transient_snes_mpfaof90.F90

```diff
@@ -185,6 +186,21 @@ program main
     endif
   endif
 
+  call DMGetDimension(dm, dim, ierr);
```

Comment:
Is this new code block related to the parallelism problem you found a week or so back? Is it something that we could pull into a function for convenience?

Reply:
This code block isn't related to the parallel problem. 

If a demo is creating a DM and using that to create TDycore, then this code block allows mapping of the local cell id to natural cell id. If TDycore is created without a DM, then TDycore does this automatically.

Reply:
Is this something we'd like to wrap in a function to make available for other demos?